apps/tangle-dapp/containers/NominationsPayoutsContainer/NominationsPayoutsContainer.tsx

```diff
@@ -110,17 +115,21 @@ const DelegationsPayoutsContainer: FC = () => {
   }, [nomineesOpt]);
 
   const fetchedPayouts = useMemo(() => {
-    if (payoutsData !== null) {
-      setPayouts(payoutsData);
-      return payoutsData;
+    if (payoutsData[maxEras]) {
+      return payoutsData[maxEras];
     }
-  }, [payoutsData]);
+
+    return [];
+  }, [payoutsData, maxEras]);
 
   useEffect(() => {
     if (updatedPayouts.length > 0) {
-      setPayouts(updatedPayouts);
+      setPayouts({
+        ...data,
```

Comment:
We don't need the spread operator here to reduce the dependency of `useEffect`, thereby preventing unnecessary re-calculations and re-renderings. Instead, we should just set the state and configure the merging behavior when defining the store.